illumina.py

```diff
@@ -182,22 +195,34 @@ def main_illumina_demux(args):
         picardOpts['num_processors'] = args.threads
     if not picardOpts.get('sequencing_center') and illumina.get_RunInfo():
         picardOpts['sequencing_center'] = illumina.get_RunInfo().get_machine()
-    tools.picard.IlluminaBasecallsToSamTool().execute(
-        illumina.get_BCLdir(),
-        barcodes_tmpdir,
-        flowcell,
-        args.lane,
-        basecalls_input,
-        picardOptions=picardOpts,
-        JVMmemory=args.JVMmemory)
+
+    if multiplexed_samples:
+        tools.picard.IlluminaBasecallsToSamTool().execute(
+            illumina.get_BCLdir(),
+            barcodes_tmpdir,
+            flowcell,
+            args.lane,
+            basecalls_input,
+            picardOptions=picardOpts,
+            JVMmemory=args.JVMmemory)
+    else:
+        tools.picard.IlluminaBasecallsToSamTool().execute_single_sample(
+            illumina.get_BCLdir(),
+            os.path.join(args.outDir,flowcell+".bam"),
+            "single_sample_"+flowcell,
```

Comment:
Open to suggestions of what to use here. The value must have a length >=5 chars per https://github.com/broadinstitute/picard/blob/2.17.6/src/main/java/picard/illumina/IlluminaBasecallsToSam.java#L510

Reply:
Right, the issue is that (as currently parsed), iSeq flowcell IDs end up as shorter than 5 chars, resulting in an exception in picard. Unfortunately there seems to be some irregularity to how the iSeq flowcells are described in `RunInfo.xml`; here are two difference schemes we've seen so far:
- `BNS417-14-15-08`
- `BNS41706-1716`

The right solution is to probably pass `flowcell` as you mention and alter our flowcell ID-parsing code by machine type, but `RunInfo.xml` does not explicitly the instrument model. We could infer it based on the `<FlowcellLayout>` attributes. So far it looks like the iSeq `<Instrument>` text is prefixed with "`FF`" (presumably for "Firefly"), so maybe we can at least remove the MiSeq-specifc check that slices based on the presence of '`-`'. Is there a historical reason for the specific bounds set on the flowcell length assertion [here](https://github.com/broadinstitute/viral-ngs/blob/v1.19.1/illumina.py#L501)?

Reply:
Hm.. great question... maybe that length check doesn't need to be quite like that, or maybe historically we discovered other Picard problems if it was over 9 characters? Certainly the lower bound should be upped to 5 based on what we're seeing in Picard.

Yeah that dash-splitting code for MiSeqs should certainly be not used in the iSeq scenario. I wonder if MiSeq machine names always start with 'M'? I wouldn't count on the iSeq always starting with 'FF' until I see data from a post-release standard machine... ours doesn't even have the model name on front (like the promo pictures do) so I could see the 'FF' changing to something else w/o notice. Or maybe they'll keep it.

I have no idea what a normal Illumina HiSeq's machine name looks like, since the only output from HiSeqs we've seen are from the Broad GP's machines and I don't know if they have their own custom naming convention. I'm pretty sure when we've run a walkup flowcell at GP we didn't get an "M###" machine ID back (maybe we can check.. I think we might have a Zika ampseq walkup MiSeq run somewhere).

How about this.. according to the [comment](https://github.com/broadinstitute/viral-ngs/blob/v1.19.1/illumina.py#L499), that string splitting is just looking for leading zeros followed by a dash in front.. maybe we should just do a proper regex on `/^0+-/` and it won't match the iSeq/FireFly dashes?